conda/gateways/repodata/jlap/interface.py

```diff
@@ -121,3 +121,28 @@ def repodata_parsed(self, state: dict | RepodataState) -> dict | None:
             raise RepodataOnDisk()
         else:
             return repodata_json_or_none
+
+    def _repodata_state_copy(self, state: dict | RepodataState):
+        return RepodataState(dict=state)
+
+
+class RepodataStateSkipFormat(RepodataState):
+    skip_formats: set[str]
+
+    def __init__(self, *args, skip_formats=set(), **kwargs):
+        super().__init__(*args, **kwargs)
+        self.skip_formats = set(skip_formats)
+
+    def should_check_format(self, format):
+        if format in self.skip_formats:
+            return False
+        return super().should_check_format(format)
+
+
+class ZstdRepoInterface(JlapRepoInterface):
+    """
+    Support repodata.json.zst (if available) without checking .jlap
+    """
+
+    def _repodata_state_copy(self, state: dict | RepodataState):
+        return RepodataStateSkipFormat(dict=state, skip_formats=["jlap"])
```

Comment:
I'm not super happy that we're implementing the zst repodata API via the JLAP API, even though they are not that related. But I don't see a clear way to implement it independently, either. Do you?

Reply:
That would include a significant refactor of [repodata/jlap/fetch](https://github.com/conda/conda/blob/aca84aa1328ca36eab1ab7f6d3956bf0a3d4b7c0/conda/gateways/repodata/jlap/fetch.py#L277-L281) which contains the logic to look for alternate formats and remember whether or not they are available.

Reply:
Gotcha, an opportunity for another time

Reply:
We also have an opportunity to add a flag or configuration to be able to disable `repodata.json.zst` checks (add both formats to the new skip-formats class), this could be written without knowledge of the networking details.